### PR TITLE
Issue 433 logs loki

### LINKED ISSUE #433: --loki flag is not working in some of gardenctl logs command
**Describe the bug**
--loki flag is not working in some of gardenctl logs command

**To Reproduce**
1. Which target was set 'gardenctl get target'
```
target:
- kind: garden
  name: dev
- kind: project
  name: sretestneo
- kind: shoot
  name: neo-test-ar
```
2. Which command was entered [e.g. 'gardenctl show vpn-seed']
`gardenctl logs api --loki`
3. What was the output of the command 
`
exit status 1
`
the backend `kubectl` command exit with status `1`
```
kubectl --kubeconfig=/Users/i352986/.garden/cache/dev/seeds/aws/kubeconfig.yaml exec loki-0 -n shoot--sretestneo--neo-test-ar -- wget 'http://localhost:3100/loki/api/v1/query_range' -O- --post-data='query={pod_name=~"kube-apiserver.*"}&&query={container_name=~"kube-apiserver.*"&&limit=200&&start=1603691324970753000&&end=1604900924970753000'
Connecting to localhost:3100 (127.0.0.1:3100)
wget: server returned error: HTTP/1.1 401 Unauthorized
command terminated with exit code 1
```
**Expected behavior**
the log should be displayed successfully

**Screenshots**
image

**Gardenctl Version (please complete the following information):**
 - latest version in master branch

**Additional context**
NA

### PR DESCRIPTION
**Pull request by andrei-panov**
**What this PR does / why we need it**:
fixes bug with fetching log using Loki
**Which issue(s) this PR fixes**:
Fixes #433 

**Special notes for your reviewer**:
target shoot cluster first and then execute:
gardenctl logs etcd-main etcd --loki

**Release note**:

```improvement
fixes the bug with fetching log using Loki and also prints more meaningful error messages if subcommands exit with an error

```

### CODE REVIEW

**Comment by jguipi** [comment]
LGTM! But can you `git rebase` before merge. And you might want @tedteng  or @neo-liang-sap approval before merge.

**Comment by dansible** [comment]
I'm getting a couple of errors here. Without the loki flag:

```
$ go run cmd/gardenctl/main.go logs api
I1119 21:09:53.613853       1 balancer_conn_wrappers.go:78] pickfirstBalancer: HandleSubConnStateChange: 0xc00c4e1ff0, {READY <nil>}
I1119 21:09:53.614511       1 controlbuf.go:508] transport: loopyWriter.run returning. connection error: desc = "transport is closing"
I1119 21:09:58.692201       1 client.go:360] parsed scheme: "passthrough"
...
```

Using the loki flag:

```
$ go run cmd/gardenctl/main.go logs --loki api
2020/11/19 16:19:41 Error from server (NotFound): pods "loki-0" not found

exit status 1
exit status 2
```

The `--loki` flag in the current master also produces an unexpected error:
```
$ go run cmd/gardenctl/main.go logs --loki api
--loki flag is available only for gardener version >= 1.8.0
Current version: 1.13.0-dev-b1e34ed2337acc6c291aa4596f5ece2fa679615c
exit status 2
```

**Comment by andrei-panov** [comment]
It's what I meant with better error messages. 
Once you got `Error from server (NotFound): pods "loki-0" not found` you could understand that there is no Loki or in our case that cluster in Hibernation most likely. Please ensure that cluster up and running during the testing.

Error with unsupported gardener version is fixed with this PR. Old implementation can't process `-dev-b1e34ed2337acc6c291aa4596f5ece2fa679615c` part. This case covered with tests now.

**Comment by dansible** [comment]
I think i may have been testing on the aws cluster which doesnt have the monitoring stack deployed (this is why the logs without the `--loki` flag were coming through but the ones for loki were not).

Can you fix the merge conflicts?

**Review by dansible — APPROVED** [review]

**Pull request merged**
